report/3.Theory/ObjectRecognition.tex

```diff
@@ -0,0 +1,146 @@
+It is not enough to simply answer the binary question "is a target present?" but rather "where is the target located?", so this information can be processed and in turn move the motors to point the laser at the target.
+As mentioned in section \ref{sec:obj_tracking}, object recognition algorithms can be tedious to do by hand, and as a result of this, machine learning algorithms will be utilized.
+Although the initial target of the device may be a simple and easily recognizable object, the intent is to create a solution that can identify more complex targets, in a potentially noisy environment.
+Because of this, it is important to look into the science of object identification, what algorithms and methods exists, and what libraries we have at our disposal.
+
+\subsection{Neural Networks}
+
+A neural network is, as the name implies, a network of neurons.
+It is based on neuroscience, especially the inner workings of the human brain, where a network of neurons is used to make decisions.
+The difference being, whether it is a biological neural network or an artificial one, which is the case in a computer science context.
+In computer science, a neuron can be seen as a function, that processes some input and then transmits the processed value to connected neurons \cite{ANN}.
+
+Neural networks are a powerful but complex method of doing machine learning.
+Because of this, simpler solutions should also be considered when exploring which methods suits the task best.
+One of the reasons neural networks are especially interesting in this case, is its ability to recognize patterns \cite{ANN}.
+Pattern recognition is a fundamental part of object recognition making neural networks an obvious option.
+
+
+\subsubsection{Convolutional Neural Network}
+Convolutional neural networks (CNN) are often used for image classification, and is the most popular type of neural network in regards to object detection\cite{CNN}.
+In challenges such as LSVCR\footnote{Imagenet Large Scale Visual Recognition Challenge} the majority of the entries are done using CNN \cite{ILSVRC_Results}.  
+A CNN is a neural network in which neurons are grouped into layers.
+These layers are grouped in three categories; an input layer, an output layer and a number of so called hidden layers.
+The goal of the hidden layers are to transform the input data into the expected output data.
+It does this by doing abstractions within each layer, breaking a complex problem into smaller, more trivial, sub-problems.
+
+In terms of object recognition, an example could be face recognition.
+The first layer could recognize lines, edges or similar low level features.
+The following layers could then detect increasingly complicated facial features, ending at some layer, in which one neuron would transmit a signal if a nose is present in the image, and another neuron to transmit if a mouth is present.
+
+
+\begin{figure}[H]
```

Comment:
Hvad med at lave denne figur om til at bruge den følgende:
\figur{0.6}{images/cnn_face}{A simplified example of the abstractions within a CNN.}{fig:face_cnn}